src/gsCore/gsDofMapper.cpp

```diff
@@ -282,6 +283,69 @@ void gsDofMapper::setIdentity(index_t nPatches, size_t nDofs)
     m_dofs.resize( m_numFreeDofs, 0);
 }
 
+void gsDofMapper::permuteFreeDofs(const gsVector<index_t>& permutation)
+{
+    GISMO_ASSERT(m_curElimId==0, "finalize() was not called on gsDofMapper");
+    GISMO_ASSERT(m_numFreeDofs == permutation.size(), "permutation size does not match number of free dofs");
+    GISMO_ASSERT(m_isPermuted==false, "you cannot permute the dofVector twice, combine the permutation");
+
+
+    //make a copy of the old ordering, easiest way to implement the permutation. Inplace reordering is quite hard.
+    std::vector<index_t> dofs = m_dofs;
```

Comment:
I though about this again, but wont work. consider the vector {1,2,3,4} and the permutation {4,3,2,1}. Then you write at the first place 4, at the second 3, but at the third again 3 and the last again 4, since you override the entries.

Reply:
Yes, you are right, now I see it, please ignore the comment